app/ca.go

```diff
@@ -0,0 +1,117 @@
+package app
+
+import (
+	"crypto/sha1"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/pem"
+	"fmt"
+	"strings"
+	"time"
+)
+
+type caCert struct {
+	Fingerprint       string    `json:"fingerprint"`
+	Issuer            string    `json:"issuer"`
+	Subject           string    `json:"subject"`
+	NotBefore         time.Time `json:"notBefore"`
+	NotAfter          time.Time `json:"notAfter"`
+	Base64EncodedData string    `json:"base64EncodedData"`
+}
+
+type caCerts []caCert
+
+func parseCertificates(encodedCerts string) (caCerts, error) {
+
+	// decode the cert bundle
+	bundle, err := base64.StdEncoding.DecodeString(encodedCerts)
+	if err != nil {
+		return nil, err
+	}
+	if len(bundle) == 0 {
+		return caCerts{}, nil
+	}
+
+	certBlocks := [][]byte{}
+	var blocks []byte
+	p := bundle
+	for {
+		var block *pem.Block
+		var rem []byte
+		block, rem = pem.Decode(p)
+		if block == nil {
+			break
+		}
+		blocks = append(blocks, block.Bytes...)
+		certBlocks = append(certBlocks, []byte(strings.TrimSpace(string(p[:len(p)-len(rem)]))))
```

Comment:
just to clarify, the purpose of this was to capture any optional headers that could have been supplied?

Reply:
This also includes the BEGIN CERTIFICATE / END CERTIFICATE, whereas block contains just the bytes within those demarcations?

Reply:
Yes that is correct. This was to keep everything in the PEM file as is.